services/github/src/utils.js

```diff
@@ -173,6 +173,94 @@ export const tailLog = (log, maxLines = 250, maxPayloadLength = 30000) => {
     .join('\n');
 };
 
+/**
+ *
+ * @param {string} logString
+ * @param {number} maxPayloadLength
+ * @param {number} tailLines
+ * @returns string or null
+ */
+export const extractTailLinesFromLog = (logString, maxPayloadLength, tailLines) => {
+
+  if(tailLines === 0) {return null;}
+
+  const tailLog = logString.split("\n").slice(-tailLines).join("\n");
+
+  if(logString.length <= maxPayloadLength) {
+    return tailLog;
+  }
+
+  let tailLogMaxPayload = logString.slice(-maxPayloadLength);
+  const newLinePosition = tailLogMaxPayload.indexOf('\n');
+  tailLogMaxPayload = tailLogMaxPayload.slice(newLinePosition + 1);
+
+  return tailLog.length <= tailLogMaxPayload.length ? tailLog : tailLogMaxPayload;
+};
+
+/**
+ *
+ * @param {string} logString
+ * @param {number} maxPayloadLength
+ * @returns string
+ */
+export const extractHeadLinesFromLog = (logString, maxPayloadLength) => {
+
+  if(logString.length <= maxPayloadLength) {return logString;}
+
+  const headLog = logString.slice(0, maxPayloadLength);
+  const lastNewLinePosition = headLog.lastIndexOf('\n');
+  return headLog.substring(0, lastNewLinePosition);
+};
+
+/**
+ * Github checks API call is limited to 64kb
+ * @param {string} log
+ * @param {number} headLines
+ * @param {number} tailLines
+ * @param {number} maxPayloadLength
+ * @returns string
+ */
+export const extractLog = (log, headLines = 20, tailLines = 200, maxPayloadLength = 30000) => {
+
+  if(headLines < 0 && tailLines < 0) {return extractLog(log, -1 * tailLines, -1 * headLines, maxPayloadLength);}
+
+  if(headLines < 0 || tailLines < 0) {
+    if(headLines < 0) {
+      return extractLog(log, 0, Math.max(tailLines, -1 * headLines), maxPayloadLength);
+    } else {
+      return extractLog(log, Math.max(headLines, -1 * tailLines), 0, maxPayloadLength);
+    }
+  }
+
+  const logString = ansi2txt(log);
+  const lines = logString.split('\n');
+
+  if(lines.length <= headLines + tailLines) {
+
+    if (maxPayloadLength <= logString.length) {
+      return extractHeadLinesFromLog(logString, maxPayloadLength);
+    }
+    return logString;
```

Comment:
within overlapping conditions, added check if the max payload length allows us to log it, ideally this condition should never be hit, but added for safety purpose, will remove if required.

Reply:
@lotas should I remove this condition as well then?

Reply:
Yeah, if lines "fit" into allowed length I think it is not necessary to do those extra manipulations we can just return all